### PR TITLE
Bug 2105325: Add ManifestListDigest field to release info struct

### PR DESCRIPTION
**Pull request by Prashanth684**
When performing oc adm release info or extract commands on heterogeneous images with tags, the release image reference is incorrect. infer the manifestlist digest and add that to the release image when referencing by tag

### CODE REVIEW

File: pkg/cli/admin/release/extract_tools.go [hunk 1]
```diff
@@ -370,12 +370,17 @@ func (o *ExtractOptions) extractCommand(command string) error {
...
 	releaseName := release.PreferredName()
 	refExact := release.ImageRef
-	refExact.Ref.Tag = ""
 	// if the release image is manifestlist image, we'll not change digest with
 	// arch based sha. Because we want that the extracted tool can be used for all archs.
 	if !release.IsManifestList {
 		refExact.Ref.ID = release.Digest.String()
+	} else {
+		// if the image is manifestlisted and referenced by tag, use the manifestlist digest.
+		if refExact.Ref.Tag != "" {
```
Comment (ardaguclu) [inline pkg/cli/admin/release/extract_tools.go:379]: I wonder if we can remove that conditional. What happened if we do this any case;

```
refExact.Ref.ID = release.ManifestListDigest
```

File: pkg/cli/admin/release/info.go [hunk 2]
```diff
@@ -1214,7 +1216,12 @@ func describeReleaseInfo(out io.Writer, release *ReleaseInfo, showCommit, showCo
 	now := time.Now()
 	fmt.Fprintf(w, "Name:\t%s\n", release.PreferredName())
 	if release.IsManifestList {
-		fmt.Fprintf(w, "Digest:\t%s\n", release.ImageRef.Ref.ID)
+		// if the image is manifestlisted and referenced by tag, use the manifestlist digest.
+		if release.ImageRef.Ref.Tag != "" {
```
Comment (ardaguclu) [inline pkg/cli/admin/release/info.go:1220]: same with the above comment. Can't we just use;

```
fmt.Fprintf(w, "Digest:\t%s\n", release.ManifestListDigest)
```

**Review by ardaguclu — COMMENTED** [review]
Thanks for the fix

File: pkg/cli/admin/release/extract_tools.go (hunk 1 above)
Comment (Prashanth684) [reply pkg/cli/admin/release/extract_tools.go:379]: it shouldn't matter as it's the same digest..i'll make the change

File: pkg/cli/admin/release/info.go [hunk 3]
```diff
@@ -692,11 +692,12 @@ type ReleaseManifestDiff struct {
...
-	ContentDigest  digest.Digest                   `json:"contentDigest"`
-	IsManifestList bool
+	Image              string                          `json:"image"`
+	ImageRef           imagesource.TypedImageReference `json:"-"`
+	Digest             digest.Digest                   `json:"digest"`
+	ContentDigest      digest.Digest                   `json:"contentDigest"`
+	IsManifestList     bool
+	ManifestListDigest string
 	// TODO: return the list digest in the future
 	// ListDigest    digest.Digest                       `json:"listDigest"`
```
Comment (aleskandro) [inline pkg/cli/admin/release/info.go:702]: @Prashanth684 is this TODO comment related to the same work you implemented here?

Comment (ardaguclu) [reply pkg/cli/admin/release/info.go:702]: This is a good point

File: pkg/cli/admin/release/extract_tools.go [hunk 4]
```diff
@@ -375,6 +375,9 @@ func (o *ExtractOptions) extractCommand(command string) error {
 	// arch based sha. Because we want that the extracted tool can be used for all archs.
 	if !release.IsManifestList {
 		refExact.Ref.ID = release.Digest.String()
+	} else {
```
[the remark this answers is not in the document]
Comment (ardaguclu) [reply pkg/cli/admin/release/extract_tools.go:378]: That means we can remove `IsManifestList` field from `ReleaseInfo`. Because this information can be extracted from `ManifestListDigest`.

[the remark this answers is not in the document]
Comment (Prashanth684) [reply pkg/cli/admin/release/extract_tools.go:378]: agreed. will make this change.

File: pkg/cli/admin/release/info.go (hunk 3 above)
Comment (Prashanth684) [reply pkg/cli/admin/release/info.go:702]: ooh...yeah it does look like that. @soltysh was this field meant to capture the manifestlist digest. if so i can just use it.

File: pkg/cli/admin/release/extract_tools.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (Prashanth684) [reply pkg/cli/admin/release/extract_tools.go:378]: fixed.

File: pkg/cli/admin/release/info.go (hunk 3 above)
Comment (Prashanth684) [reply pkg/cli/admin/release/info.go]: removed TODO and made the manifestlistdigest a digest.Digest type

**Review by soltysh — APPROVED** [review]
/lgtm
/approve
